alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/clustering/ClusteringMetricsBeanPostProcessor.java

```diff
@@ -0,0 +1,72 @@
+package eu.xenit.alfred.telemetry.binder.clustering;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.config.ConstructorArgumentValues;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+
+public class ClusteringMetricsBeanPostProcessor implements BeanDefinitionRegistryPostProcessor {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClusteringMetricsBeanPostProcessor.class);
+
+    public static final String CLUSTER_SERVICE = "clusterService";
+    public static final String CLUSTERING_METRICS_ENABLED_PROPERTY = "alfred.telemetry.binder.clustering.enabled";
+    public static final String CLUSTERING_METRICS_BEAN_ID = "eu.xenit.alfred.telemetry.binder.clustering.ClusteringMetrics";
+    public static final String COMMUNITY_CLUSTERING_METRICS_BEAN_ID = "eu.xenit.alfred.telemetry.binder.clustering.CommunityClusteringMetrics";
+
+    private final Properties globalProperties;
+
+    private MeterRegistry meterRegistry;
+
+    public ClusteringMetricsBeanPostProcessor(Properties globalProperties, MeterRegistry meterRegistry) {
+        this.globalProperties = globalProperties;
+        this.meterRegistry = meterRegistry;
+    }
+
+    @Override
+    public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry beanDefinitionRegistry) throws BeansException {
+        createClusteringMetricsBean(beanDefinitionRegistry);
+    }
+
+    private void createClusteringMetricsBean(BeanDefinitionRegistry beanDefinitionRegistry) {
+        if (!Boolean.parseBoolean(globalProperties.getProperty(CLUSTERING_METRICS_ENABLED_PROPERTY))) {
```

Comment:
This logic is already present in the `MeterBinderRegistrar`. There, the `ClusteringMetrics` will automatically be enabled / disabled based on the `alfred.telemetry.binder.clustering.enabled` property. 

Therefore, I'd prefer to remove it here to avoid confusion.

Reply:
Aha, this would mean the property for `CommunityClusteringMetrics` would be different (`alfred.telemetry.binder.community-clustering.enabled`). 

You could make `CommunityClusteringMetrics` implement `NamedMeterBinder` and force the name of these metrics to `clustering` too.

Reply:
Very good point, pushed a change for this